webdriver/navigation.py

```diff
@@ -1,109 +1,96 @@
 import json
 import pytest
 import types
-import urllib
-
-import webdriver
-
-
-def inline(doc):
-    return "data:text/html;charset=utf-8,%s" % urllib.quote(doc)
 
+from util.inline import inline
+from util import wd_assert
 
 alert_doc = inline("<script>window.alert()</script>")
 frame_doc = inline("<p>frame")
 one_frame_doc = inline("<iframe src='%s'></iframe>" % frame_doc)
 two_frames_doc = inline("<iframe src='%s'></iframe>" % one_frame_doc)
 
 
-@pytest.fixture
-def new_window(session):
-    """Open new window and return the window handle."""
-    windows_before = session.window_handles
-    name = session.execute_script("window.open()")
-    assert len(session.window_handles) == len(windows_before) + 1
-    new_windows = session.window_handles - windows_before
-    return new_windows.pop()
-
-
 # TODO(ato): 7.1 Get
 
 
-def test_get_current_url_no_browsing_context(session, new_window):
+def test_get_current_url_no_browsing_context(session, create_window):
     # 7.2 step 1
-    session.window_handle = new_window
+    session.window_handle = create_window()
     session.close()
-    with pytest.raises(webdriver.NoSuchWindowException):
-        session.url = "about:blank"
+
+    result = session.send_raw_command("GET", "url")
+
+    wd_assert.error(result, "no such window")
```

Comment:
This change makes the test more direct. The current factoring "reads" more like
a test for the `wdclient` library than for the specification itself--something
like, "Setting the `url` property on the `wdclient.Client` instance should
raise a Python exception." It also places more non-critical logic in the tested
code path.

My hope is that expressing the test subject in terms of `send_raw_command` will
make the test more legible to people who are familiar with the specification
but not the W3C's library--"Sending this command should return an error."